pkg/cmd/server/origin/master.go

```diff
@@ -692,3 +693,34 @@ type clientDeploymentInterface struct {
 func (c clientDeploymentInterface) GetDeployment(ctx api.Context, name string) (*api.ReplicationController, error) {
 	return c.KubeClient.ReplicationControllers(api.NamespaceValue(ctx)).Get(name)
 }
+
+// namespacingFilter adds a filter that adds the namespace of the request to the context.  Not all requests will have namespaces,
+// but any that do will have the appropriate value added.
+func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMapper) http.Handler {
+	infoResolver := &apiserver.APIRequestInfoResolver{util.NewStringSet("api", "osapi"), latest.RESTMapper}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx, ok := contextMapper.Get(req)
+		if !ok {
+			http.Error(w, "Unable to find request context", http.StatusInternalServerError)
+			return
+		}
+
+		if _, exists := kapi.NamespaceFrom(ctx); !exists {
+			requestInfo, _ := infoResolver.GetAPIRequestInfo(req)
```

Comment:
still feels weird to set an empty namespace on the context regardless of the success of GetAPIRequestInfo.

If GetAPIRequestInfo() fails, and requestInfo.Namespace is empty, what does setting an empty namespace communicate to downstream consumers of the context?